spacy/strings.pyx

```diff
@@ -123,127 +32,120 @@ cdef class StringStore:
+        """If a string is passed as the input, return it as-is. If the input
+        is a hash value, return its corresponding string.
 
-        string_or_id (str or int): The string to check.
-        RETURNS (bool): Whether the store contains the string.
+        string_or_hash (str, int): The hash value to lookup or a string.
+        RETURNS (str): The stored string or the input string.
         """
-        cdef hash_t str_hash
-        if isinstance(string_or_id, str):
-            if len(string_or_id) == 0:
-                return True
-            elif string_or_id in SYMBOLS_BY_STR:
-                return True
-            str_hash = hash_string(string_or_id)
-        elif _try_coerce_to_hash(string_or_id, &str_hash):
-            pass
+        if isinstance(string_or_hash, str):
+            return string_or_hash
         else:
-            # TODO: Raise an error instead
-            return self._map.get(string_or_id) is not NULL
+            return self._get_interned_str(string_or_hash)
 
-        if str_hash in SYMBOLS_BY_INT:
-            return True
-        else:
-            return self._map.get(str_hash) is not NULL
+    def items(self) -> Collection[Tuple[str, int]]:
+        """Iterate over the stored strings and their hashes in insertion order.
+
+        RETURNS: An iterable collection of string-hash pairs.
+        """
+        # Even though we internally store the hashes as keys and the strings as
+        # values, we invert the order in the public API to keep it consistent with
+        # the implementation of the `__iter__` method (where we wish to iterate over
+        # the strings in the store).
+        def as_pairs(string: str, str_hash: int) -> Tuple[str, int]:
+            return (string, str_hash)
 
-    def __iter__(self):
-        """Iterate over the strings in the store, in order.
+        return self._transform_to_list(as_pairs)
 
-        YIELDS (str): A string in the store.
+    def keys(self) -> Collection[str]:
+        """Iterate over the stored strings in insertion order.
+
+        RETURNS: An iterable collection of strings.
         """
-        cdef int i
-        cdef hash_t key
-        for i in range(self.keys.size()):
-            key = self.keys[i]
-            utf8str = <Utf8Str*>self._map.get(key)
-            yield decode_Utf8Str(utf8str)
-        # TODO: Iterate OOV here?
+        def as_strings(string: str, str_hash: int) -> str:
+            return string
 
-    def __reduce__(self):
-        strings = list(self)
-        return (StringStore, (strings,), None, None, None)
+        return self._transform_to_list(as_strings)
+
+    def values(self) -> Collection[int]:
+        """Iterate over the stored strings hashes in insertion order.
+
+        RETURNS: An iterable collection of string hashs.
+        """
+        def as_hashes(string: str, str_hash: int) -> int:
+            return str_hash
+
+        return self._transform_to_list(as_hashes)
```

Comment:
Is there any reason you can't just iterate over `_keys` here instead?

Reply:
Good point, I overlooked that.